crates/neon/src/types_impl/error.rs

```diff
@@ -11,7 +11,30 @@ use crate::{
     types::{build, private::ValueInternal, utf8::Utf8, Value},
 };
 
-/// A JS `Error` object.
+/// The type of JavaScript
+/// [`Error`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error)
+/// objects.
+///
+/// # Example
+///
+/// ```
+/// # use neon::prelude::*;
+/// # fn test(mut cx: FunctionContext) -> JsResult<JsUndefined> {
+/// // Create a type error:
+/// let err = cx.type_error("expected a number, found a string")?;
+///
+/// // Add some custom diagnostic properties to the error:
+/// let expected = cx.string("number");
+/// err.set(&mut cx, "expected", expected)?;
```

Comment:
This looks good, but this makes me wonder if there's some API tweak we could make so that you don't need the `expected` temporary. For example, if `cx` was the *last* argument, it wouldn't be necessary, but that's also strange.

Reply:
Yeah, it's a shame, and I've had the same thought. I'm not sure I have a suggestion, since `cx` as the first argument is pretty baked in to Neon at this point.